### PR TITLE
 add reduce_grid methods in the cart and tensorfem

### PR DESCRIPTION
**Pull request by saidctb**

### CODE REVIEW

File: psydac/fem/tensor.py [hunk 1]
```diff
@@ -312,6 +341,14 @@ def local_domain( self ):
         """
         return self._element_starts, self._element_ends
 
+    @property
+    def local_element_starts( self ):
```
Comment (yguclu) [inline psydac/fem/tensor.py]: Shouldn't this property be called `global_element_starts`?

File: psydac/fem/tensor.py [hunk 2]
```diff
@@ -312,6 +341,14 @@ def local_domain( self ):
         """
         return self._element_starts, self._element_ends
 
+    @property
+    def global_element_starts( self ):
+        return self._global_element_starts
+
+    @property
+    def local_element_ends( self ):
```
Comment (yguclu) [inline psydac/fem/tensor.py]: Just like the previous property, this one should be called `global_element_ends`.

File: psydac/ddm/cart.py [hunk 3]
```diff
@@ -332,7 +332,54 @@ def remove_last_element( self, axes):
 
         return cart
 
+    def reduce_grid(self, global_starts, global_ends):
```
[the remark this answers is not in the document]
Comment (saidctb) [reply psydac/ddm/cart.py]: done

File: psydac/fem/tensor.py [hunk 4]
```diff
@@ -363,6 +400,43 @@ def compute_interpolant( self, values, field ):
             out     = field.coeffs,
         )
 
+    def reduce_grid(self, axes=[], knots=[]):
```
[the remark this answers is not in the document]
Comment (saidctb) [reply psydac/fem/tensor.py]: i have added the test

File: psydac/fem/tensor.py (hunk 2 above)
Comment (saidctb) [reply psydac/fem/tensor.py]: thanks i forgot to change it

File: psydac/fem/tensor.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (saidctb) [reply psydac/fem/tensor.py]: I have added it

**Pull request merged**